lib/Common/Memory/HeapBucket.h

```diff
@@ -267,6 +268,10 @@ template <bool pageheap, typename Fn>
 void
 HeapBucketT<TBlockType>::SweepBucket(RecyclerSweep& recyclerSweep, Fn sweepFn)
 {
+    if (recyclerSweep.IsBackground()) 
```

Comment:
changed to just reuse the existing uncollectedNewPageCount, I tried with this the perf and fragmentation seems similar to before the change